src/openrct2-ui/windows/Park.cpp

```diff
@@ -1446,6 +1446,22 @@ static void window_park_objective_invalidate(rct_window* w)
     window_park_anchor_border_widgets(w);
 }
 
+static void window_park_objective_bars_paint(
+    float percentage, int32_t x, int32_t y, rct_window* w, rct_drawpixelinfo* dpi, int32_t colour)
+{
+    if (percentage > 1)
+    {
+        percentage = 1;
+    }
+    int16_t barLength = static_cast<int16_t>((221 - 4) * percentage);
+
+    gfx_fill_rect_inset(dpi, x + 1, y + 1, x + 221 - 3, y + 9, w->colours[1], INSET_RECT_F_30);
+    if (barLength > 2)
+    {
+        gfx_fill_rect_inset(dpi, x + 3, y + 2, x + barLength, y + 8, colour, 0);
```

Comment:
Same as above.

Reply:
I've switched to using ScreenRects now. :)